### ISSUE TITLE
Remove LAPACK as a requirement

### ISSUE THREAD

**Issue by Connor-Bowley** [Kitware/SlicerSALT#282]
Is it worthwhile to remove LAPACK (and by extension Fortran) for an easier build process and lower barrier to development.

GROUPS doesn't appear to need LAPACK (https://github.com/NIRALUser/GROUPS/issues/40).

SPHARM-PDM uses it in one place that I have found. Could we replace the LAPACK call with something equivalent from Eigen (which is already in the superbuild and is used by other things like VTK) and then remove LAPACK (and Fortran) altogether?

**Comment by bpaniagua** [comment]
yes, that would be fine, but we will have to benchmark SPHARM and make sure it continues working the same with the same dataset

**Comment by jcfr** [comment]
> SPHARM-PDM uses it

For reference, the lapack function `LAPACK_sgels` is used in `ParametricMeshToSPHARMSpatialObjectFilter::ComputeCoeffs`

  https://github.com/NIRALUser/SPHARM-PDM/blob/47188484cc4e480ea725d37d997c7b1a6e143b14/Libraries/Shape/Algorithms/ParametricMeshToSPHARMSpatialObjectFilter.cxx

Related links:
* https://netlib.org/lapack/explore-html/d0/db8/group__real_g_esolve_ga166c189c7bcd808a9468c05e53da816f.html
* https://github.com/Reference-LAPACK/lapack/blob/77a0ceb6c9a3c757e7039027f1f324e2811e6ee0/SRC/sgels.f

**Comment by jcfr** [comment]
Closing. Both SPHARM-PDM and GROUPS were respectively updated by @allemangD in the following pull-requests:
* https://github.com/NIRALUser/SPHARM-PDM/pull/81
* https://github.com/NIRALUser/GROUPS/pull/44

And corresponding updates integrated into SlicerSALT through https://github.com/Kitware/SlicerSALT/pull/294
